### ISSUE TITLE
Mesh_3: Save/load c3t3 objects correctly

### ISSUE THREAD

**Issue by lrineau** [CGAL/cgal#2359]
## Issue Details

For the moment, the I/O of C3t3 objects only save/load the information of the TDS_3, and the I/O of `C3t3::Triangulation::Vertex` and `C3t3::Triangulation::Cell` ensures that most of the information of the C3t3 is saved and restored.

But a C3t3 object has data that is not stored in the TDS. Some of it can be retrieved after a load by a call to the *undocumented* method `c3t3.rescan_after_load_of_triangulation()`, but the `c3t3.is_in_complex(Edge)` information cannot be retrieved. It should be stored in the file, and loaded from it.

This issue will require a solution to #861 first.

## Environment

* CGAL version: CGAL-4.11

Cc: @sloriot

**Comment by sloriot** [comment]
Shall this issue be handled by @maxGimeno ?

**Comment by lrineau** [comment]
The issue pointed by #861 is a real blocker for this one.

**Comment by sloriot** [comment]
To me it seems this issue has a proposal solution :)

**Comment by maxGimeno** [comment]
Indeed, you seem to have agreed with Sebastien's proposal to add new functions to complete the IO. I could probably take care of both issues.

**Comment by janetournois** [comment]
related to issue #7780

**Comment by janetournois** [comment]
PR #7821 solves this issue
